crates/matrix-sdk/build.rs

```diff
@@ -0,0 +1,37 @@
+use std::{env, process};
+
+fn env_is_set(var_name: &str) -> bool {
+    env::var_os(var_name).is_some()
+}
+
+fn ensure(cond: bool, err: &str) {
+    if !cond {
+        eprintln!("error: {err}");
+        process::exit(1);
+    }
+}
+
+fn main() {
+    let native_tls_set = env_is_set("CARGO_FEATURE_NATIVE_TLS");
+    let rustls_tls_set = env_is_set("CARGO_FEATURE_RUSTLS_TLS");
+    ensure(
+        native_tls_set || rustls_tls_set,
+        "one of the features 'native-tls' or 'rustls-tls' must be enabled",
+    );
+    ensure(
+        !native_tls_set || !rustls_tls_set,
+        "only one of the features 'native-tls' or 'rustls-tls' can be enabled",
+    );
+
+    let is_wasm = env::var_os("CARGO_CFG_TARGET_ARCH").map_or(false, |arch| arch == "wasm32");
+    if is_wasm {
+        ensure(
```

Comment:
Do we want to emit the error when someone tries to use an unsupported store as well? That is, the sled store under wasm and the indexeddb store under non-wasm targets?

Reply:
that would have to happen in the corresponding crate, but is probably a good idea...